File: src/main/java/com/mizu20040814/kakeiboapi/entity/Expense.java
package com.mizu20040814.kakeiboapi.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Table(name="expenses")
@Getter
@Setter
@NoArgsConstructor

public class Expense {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private LocalDate date;

    @Column(nullable = false)
    private Integer amount;

    @Column(nullable = false,length = 50)
    private String category;

    @Column(length = 200)
    private String memo;

}

File: frontend/types/expense.ts
export type Expense = {
    id: number;
    date: string;
    amount: number;
    category: string;
    memo: string;
};

export type ExpenseFormData = {
    date: string;
    amount: number;
    category: string;
    memo: string;
};

export const CATEGORIES = [
    "食費",
    "交通費",
    "日用品",
    "娯楽",
    "その他",
] as const;

export type Category = (typeof CATEGORIES)[number];

export const CATEGORY_ICONS: Record<Category, string> = {
    食費: "🍽️",
    交通費: "🚃",
    日用品: "🧴",
    娯楽: "🎮",
    その他: "📦",
};
